Fix spelling error in `contains_branch_separately` method name

brchristian opened the pull request:
Fixes #30474.

## PR summary

This PR fixes a long-standing spelling error in the `contains_branch_seperately` method name, which should be `contains_branch_separately`.

In order to maintain backwards compatibility, aliases maintain the old spelling for legacy use. (Per discussion in #30474 with @tacaswell it was reasoned that this was preferable to deprecation.)

Type hints and internal references are updated to the new spelling accordingly.

## PR checklist
- [x] "closes #0000" is in the body of the PR description to [link the related issue](https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue)
- [x] new and changed code is [tested](https://matplotlib.org/devdocs/devel/testing.html)
- [N/A] *Plotting related* features are demonstrated in an [example](https://matplotlib.org/devdocs/devel/document.html#write-examples-and-tutorials)
- [N/A] *New Features* and *API Changes* are noted with a [directive and release note](https://matplotlib.org/devdocs/devel/api_changes.html#announce-changes-deprecations-and-new-features)
- [N/A] Documentation complies with [general](https://matplotlib.org/devdocs/devel/document.html#write-rest-pages) and [docstring](https://matplotlib.org/devdocs/devel/document.html#write-docstrings) guidelines

tacaswell commented on lib/matplotlib/transforms.pyi:
```diff
@@ -189,7 +189,10 @@ class Transform(TransformNode):
+    def contains_branch_separately(
+        self, other_transform: Transform
+    ) -> Sequence[bool]: ...
+    def contains_branch_seperately(  # Alias (historical typo)
```
Agree with dropping the wrong spelling from the typehinting.

QuLogic commented:
I'll also rename the PR title, as people tend to want to open PRs to 'fix' the changelog on intentional typos.

brchristian commented:
Thanks for the review, and great points!

I've updated the alias to an actual method, like so:

```python
    def contains_branch_seperately(self, other_transform):
        """Permanent alias for backwards compatibility (historical typo)."""
        return self.contains_branch_separately(other_transform)
```

As you note, this means we no longer need additional aliases in child classes or mixins.

When I tried to remove the alias completely from `transforms.pyi`, the CI tests failed. But we should be able to do it pretty minimally. Instead of duplicating the type hints, I simplified that down to a single line:

```python
contains_branch_seperately = contains_branch_separately  # Alias (historical typo)
```

And I removed the code from `BlendedMixin` in `transforms.pyi`.

That seems to make CI happy, and I think that's about as minimal as it can get in `transforms.pyi`, but of course LMK if you prefer a different solution. (One alternative could be a `@no_type_check` decorator in `transforms.py`, for instance.) I'm of course totally happy to change it up if you prefer something else.

QuLogic commented:
> When I tried to remove the alias completely from `transforms.pyi`, the CI tests failed. But we should be able to do it pretty minimally. Instead of duplicating the type hints, I simplified that down to a single line:

Ah, good point, I forgot we check for consistency. This is fine as it is.

QuLogic commented on lib/matplotlib/transforms.py:
```diff
@@ -1418,24 +1418,28 @@ def contains_branch(self, other):
         # just return the appropriate shape.
         return (self.contains_branch(other_transform), ) * 2
 
+    def contains_branch_seperately(self, other_transform):
+        """Permanent alias for backwards compatibility (historical typo)."""
```
If you set this private, then it shouldn't be documented (as I believe was decided in #30474)
```suggestion
    # Permanent alias for backwards compatibility (historical typo).
    def contains_branch_seperately(self, other_transform):
        """:meta private:"""
```

brchristian replied:
Great point! Done.
